src/Features/LanguageServer/Protocol/CustomProtocol/FindUsagesLSPContext.cs

```diff
@@ -275,7 +275,10 @@ public override async ValueTask OnReferenceFoundAsync(SourceReferenceItem refere
         private Task ReportReferencesAsync(ImmutableArray<VSReferenceItem> referencesToReport, CancellationToken cancellationToken)
         {
             // We can report outside of the lock here since _progress is thread-safe.
-            _progress.Report(referencesToReport.ToArray());
+            foreach (var reference in referencesToReport)
+            {
+                _progress.Report(reference);
+            }
```

Comment:
`BufferedProgress` will take each of these items, and wrap it in its own array. Presumably fixing this will make pull diagnostics have to jump through similar hoops though, so I don't know if it really matters. Thoughts?

Either way, I thought it best to not change it now, to avoid conflicts with the pull diagnostics PR.

Reply:
yeah, we should likely allow unifying this to avoid the double wrapping.  i can look to that tomorrow.